src/diagnostic/encode.ts

```diff
@@ -19,7 +19,7 @@ export default function(type: Event): Token[] {
             break;
         case Event.ImageError:
             tokens.push(image.data.source);
-            tokens.push(observe(image.data.target));
+            tokens.push(observe(image.data.target as Node));
```

Comment:
>observe [](start = 24, length = 7)

How does this get fixed up if the target is not known yet and the observe function returns null? Put the question here but it is a generic one for the entire approach.